models/cycle_gan_sty2_model.py

```diff
@@ -264,19 +267,19 @@ def forward(self):
             #self.netDecoderG_B.eval()
             if self.rec_noise > 0.0:
                 self.fake_B_noisy1 = self.gaussian(self.fake_B, self.rec_noise)
-                self.z_rec_A, self.n_rec_A = self.netG_B(self.fake_B_noisy1)
+                self.z_rec_A, self.n_rec_A = self.netG_B(nnf.interpolate(self.fake_B_noisy1, size=(self.opt.crop_size), mode='bicubic', align_corners=False))
```

Comment:
Maybe only interpolate if the image size is not already the one requested ? Or maybe this is what `nnf.interpolate` does already, i.e. does nothing if size is correct ?

Reply:
Also, you may want to separate the interpolation call from the `netG` and other calls, as it makes the code difficult to read.

Reply:
I modified the code according to your advices :)